src/Microsoft.Extensions.CodeGenerators.Mvc/Controller/ControllerWithContextGenerator.cs

```diff
@@ -26,8 +40,87 @@ public abstract class ControllerWithContextGenerator : ControllerGeneratorBase
             EntityFrameworkService = entityFrameworkService;
         }
 
+        public override async Task Generate(CommandLineGeneratorModel controllerGeneratorModel)
+        {
+            Contract.Assert(!String.IsNullOrEmpty(controllerGeneratorModel.ModelClass));
+
+            string outputPath = ValidateAndGetOutputPath(controllerGeneratorModel);
+            var modelTypeAndContextModel = await ValidateModelAndGetMetadata(controllerGeneratorModel);
+
+            if (string.IsNullOrEmpty(controllerGeneratorModel.ControllerName))
+            {
+                //Todo: Pluralize model name
+                controllerGeneratorModel.ControllerName = modelTypeAndContextModel.ModelType.Name + Constants.ControllerSuffix;
+            }
+
+            var templateModel = new ControllerWithContextTemplateModel(modelTypeAndContextModel.ModelType, modelTypeAndContextModel.DbContextFullName)
+            {
+                ControllerName = controllerGeneratorModel.ControllerName,
+                AreaName = string.Empty, //ToDo
+                UseAsync = controllerGeneratorModel.UseAsync,
+                ControllerNamespace = GetControllerNamespace(),
+                ModelMetadata = modelTypeAndContextModel.ModelMetadata
+            };
+
+            await CodeGeneratorActionsService.AddFileFromTemplateAsync(outputPath, GetTemplateName(controllerGeneratorModel), TemplateFolders, templateModel);
+            Logger.LogMessage("Added Controller : " + outputPath.Substring(ApplicationEnvironment.ApplicationBasePath.Length));
+
+            await GenerateViewsIfRequired(controllerGeneratorModel, modelTypeAndContextModel, templateModel.ControllerRootName);
+        }
+
+        private async Task GenerateViewsIfRequired(CommandLineGeneratorModel controllerGeneratorModel,
+            ModelTypeAndContextModel modelTypeAndContextModel,
+            string controllerRootName)
+        {
+            if (!controllerGeneratorModel.IsRestController && !controllerGeneratorModel.NoViews)
+            {
+                var layoutDependencyInstaller = ActivatorUtilities.CreateInstance<MvcLayoutDependencyInstaller>(ServiceProvider);
+                await layoutDependencyInstaller.Execute();
+
+                foreach (var viewTemplate in _views)
+                {
+                    var viewName = viewTemplate == "List" ? "Index" : viewTemplate;
```

Comment:
The template name for 'Add View' dialog is 'List', I am not sure if there was some background, I kind of took the same list as VS 2013, perhaps the view name List makes more sense when generating individual view which shows all entities from database but when generating several views for controller we want the view name to be 'index' because that's the default.

If there is no push back , we could actually change the list of template names for view generator to have 'Index' as well instead of 'List'